src/zenml/stack/stack.py

```diff
@@ -68,6 +71,8 @@
 
 logger = get_logger(__name__)
 
+_STACK_CACHE: Dict[Tuple[UUID, Optional[datetime]], "Stack"] = {}
```

Comment:
Actually we often call `Stack.from_model(...)` with specific IDs while running a step, which is why this cache was necessary.

Reply:
I see how this helps here, but I have a follow-up question too. Going forward, wouldn't it be cleaner to use `client.active_stack` in places where we use `Stack.from_model(...)` instead of the cache here? or am I missing something that doesn't allow us to do that?